docs/docs/architecture.md

```diff
@@ -122,3 +122,50 @@ to demonstrate ways of organising different projects.
           - iam-role-admin.yaml
           - iam-role-developer.yaml
   ```
+
+- Per project configurations sharing the same templates. This is useful if you
+  have common shared templates that are shared between many projects and would
+  like to encourage template reuse. In this example, vpc.yaml and database.yaml
+  are common templates, and each deployed environment should have its own
+  configured VPC and database.
+
+  ```
+  - templates
+      - application_1
+          - application_1.yaml
+      - application_2
+          - application_2.yaml
+      - shared
+          - vpc.yaml
+          - database.yaml
+  - config
+      - project_1
+          - dev
+              - vpc.yaml        
+              - database.yaml      
+              - application_1.yaml
+          - qa
+              - vpc.yaml        
+              - database.yaml      
+              - application_1.yaml
+          - prod
+              - vpc.yaml        
+              - database.yaml      
+              - application_1.yaml
+      - project_2
+          - dev
+              - vpc.yaml        
+              - database.yaml      
+              - application_2.yaml
+          - qa
+              - vpc.yaml        
+              - database.yaml      
+              - application_2.yaml
+          - prod
+              - vpc.yaml        
+              - database.yaml      
+              - application_2.yaml              
+  ```
+
+To deploy project_1 to dev, you would run
+`sceptre --config-dir config/project_1 launch dev`
```

Comment:
@pwrmiller how do you see this being different to: 

`sceptre launch project_1/dev` or `sceptre --ignore-dependencies launch project_1/dev`

In the the case above you would need to make sure that there were no implicit or explicit dependencies between any of the project configs otherwise sceptre will load them (unless you use `--ignore-dependencies`).

I think that the main thing here is that if you needed to call your config directory something other than `config` you could do so with the `--config-dir` flag.

Another thought is that if you have the same configs that are in dev/test/prod set up you'd ideally want to have one set of files that are deployed as different stacks, which can be achieved with symbolic links.

Reply:
This is a little different than `sceptre launch project1/dev` (in fact, if you had the layout in this example, that command results in an error if there are dependencies, as you raised). 

You're right that there shouldn't be implicit or explicit dependencies between any project configs - this would produce completely independent project environments (e.g. dev/ qa/ prod) from the same set of templates but different config. 

This is desirable for those who like to stand up isolated environments (e.g. separate VPC per project per environment), but where the configuration may differ slightly (e.g. number of AZs).

We could get around this by passing a vars file and using symlinks, but having separate configurations for each project allows for more flexibility in terms of variable configuration without having to pass in var files / templating our stack group configurations. It also allows arbitrary nesting in the config directory (in this example used to define multiple independent environments for multiple projects). 

Symlinks are also sometimes an issue on Windows; and although I don't use that platform, we should be cognizant that other users of sceptre might.